codex/slots/proofs/backends/circomcompat.nim

```diff
@@ -0,0 +1,237 @@
+## Nim-Codex
+## Copyright (c) 2024 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+{.push raises: [].}
+
+import std/sequtils
+
+import pkg/chronos
+import pkg/questionable/results
+import pkg/circomcompat
+import pkg/poseidon2/io
+
+import ../../types
+import ../../../stores
+import ../../../merkletree
+
+import pkg/constantine/math/arithmetic
+import pkg/constantine/math/arithmetic
+import pkg/constantine/math/io/io_bigints
```

Comment:
Do you follow any conventions for grouping/ordering imports? Would be nice to know which so I can follow the same.

Reply:
ah, good catch I just saw it as well... 

no strict conventions and constantine imports are quite skewed